sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -199,7 +201,12 @@ boolean isContentResponseOnWriteEnabled() {
      * an error.
      */
     public Mono<CosmosAsyncDatabaseResponse> createDatabaseIfNotExists(CosmosDatabaseProperties databaseSettings) {
-        return createDatabaseIfNotExistsInternal(getDatabase(databaseSettings.getId()));
+        if(!getTracerProvider().isEnabled()) {
+            return createDatabaseIfNotExistsInternal(getDatabase(databaseSettings.getId()));
+        }
+
+        return withContext(context -> createDatabaseIfNotExistsInternal(getDatabase(databaseSettings.getId()),
```

Comment:
The `withContext` overload would be used for context passing and not just tracing context-related so should not be opted in for regardless of tracing presence.
@srnagar

Reply:
We don't want any perf hit due to tracing code flow if user has not opt in the tracing. Unless we have any requirement which need with context , and that will be out of scope of this PR

Reply:
as discussed offline will introduce this on demand

Reply:
I would like this to be resolved within this PR - Context is general purpose and not specifically tied to tracing. That it is treated as such here means that we are not following convention and users who are using Context for other purposes, e.g. in their own http policies, will get a surprise when the context is not properly propagating for them.

Reply:
With context having perf hit , if we do it on all, then non tracer user will get affected , already communicated to Sameeksha and Srikanta. We will track it as separate item and will introduce it on demand and we will then revisit perf again.